lib/Home.dart

```diff
@@ -116,23 +121,50 @@ class _HomeState extends State<Home> {
     );
   }
 
+  Widget _profilePage(BuildContext context) {
+    return Column (
+      children: <Widget>[
+        Profile(),
+        Padding (
+          padding: EdgeInsets.only(top: 6),
+          child: AccountInfo()
+        )
+      ],
+    );
+  }
+  Widget getPage(BuildContext context, int index) {
+    switch (index) {
+      case (RIDES):
+        if (rides.length == 0) return _noRidesLeftPage(context);
+        else return _ridesPage(context);
+        break;
+      case (HISTORY):
+        return Column(
+          children: <Widget>[
+            SizedBox(height: 50),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                SignOutButton(),
+              ],
+            )
+          ],
+        );
+        break;
+      case (PROFILE):
+        return _profilePage(context);
+        break;
+      default:
+        return Column();
+    }
+  }
   @override
   Widget build(BuildContext context) {
+    rides.add(UpcomingRide());
```

Comment:
Could you please explain the purpose of always (unconditionally) adding to rides during the build? Would this not cause constant re-rendering?

You can just initialize rides beforehand, using initState (which will most likely be used for fetching in the future).

Reply:
Shoot, sorry! This was a temporary line I had to get the rides page to show up.

Reply:
Ok! So I think I changed it to make the rides page visible on master. So if this branch is up to date with master, you should be able to see the rides page without this line.